### ISSUE TITLE
DateFormats "Week of Year"  is displayed wrong

### ISSUE THREAD

**Issue by firedfly2** [amcharts/amcharts4#2712]
**Bug description**

I set "Week of Year (YYYY)" to dateFormats of DateAxis ([CodePen](https://codepen.io/firedfly/pen/dyGQPrK)).

```
dateAxis.dateFormats.setKey("year", "yyyy/M/d(EEE) -> YYYY");
```

I got this this result.

- 2018/1/1(Tue) -> 2018
- 2019/1/1(Tue) -> **2018**
- 2020/1/1(Wed) -> **2019**

![image](https://user-images.githubusercontent.com/4607488/87813011-0bfffa80-c89c-11ea-89fb-da611855d62d.png)

This result is contrary to the [ISO week date](https://en.wikipedia.org/wiki/ISO_week_date) definition.
According to the ISO definition, 1/1 should only be treated as the previous year if 1/1 is a Friday, Saturday or Sunday.

**Environment (if applicable)**
- amcharts4: 4.9.2
- Google Chrome: 83.0.4103.116

**Comment by martynasma** [comment]
January 1st 2019 is Tuesday, hence that week belongs to 2018, hence 2018 in the label.

Same thing with January 1st, 2020 - it's Wednesday and belongs to the previous year.

**Comment by firedfly2** [comment]
@martynasma 
Thank you for replying.

"Week of Year" must be Friday, Saturday and Sunday in order for it to be the previous year.
I think the correct response is this.

2005-01-01(Sat) -> **2004**-W53-6
2006-01-01(Sun) -> **2005**-W52-7
2007-01-01(Mon) -> 2007-W01-1

[![image](https://user-images.githubusercontent.com/4607488/87843205-9f155080-c8ed-11ea-8f91-96123b2730ac.png)](https://en.wikipedia.org/wiki/ISO_week_date)
https://en.wikipedia.org/wiki/ISO_week_date

**Comment by martynasma** [comment]
Alright, thanks. I'll need to do some thinking about this :)

Will get back to you.

**Comment by martynasma** [comment]
This issue was fixed in dev version, and will be released soon.

We'll notify you here when that happens.

Thanks for bringing this to our attention!

**Comment by martynasma** [comment]
Fixed in [4.9.31](https://github.com/amcharts/amcharts4/releases/tag/4.9.31).

## [4.9.31] - 2020-07-21

### Changed
- Global adapters are being deprecated due to performance reasons. They will still work on some properties, but may be turned off at some future version. If you are using `am4core.globalAdapter`, please consider refactoring your code in some other way.
- If `groupInterval` is manually set, `DateAxis` now prepares all intervals. Normally, the chart does not prepare data sets for time intervals longer than the span of dates in the data. This causes problem in case user sets these intervals manually using `groupInterval`. Note, the groupInterval must be set initially, before data parsing in order data to be grouped into all groupIntervals.

### Fixed
- "Year of week" (`YYYY`) was not being calculated correctly by `DateFormatter`.
- Regression plugin was not working correctly in an all-`ValueAxis` scenarios.
- `ColumnSeries` were not working with `dropFromPreviousState`.
- If data was changed on a chart which was zoomed and there were less data items in the new data than there was before, a JS error was triggered.
- Sometimes a chart with `XYChartScrollbar` with series in it could produce a JS error if series has its `fillOpacity > 0`.
- Series in an `XYChartScrollbar` was not being shown if `options.onlyShowOnViewport = true`.
- `zoomToRectangle()` call was resulting in an error if called after the `MapChart` was rotated.
- `CurvedColumnSeries` columns were not being filled with proper color (black was being used instead).
- A chart with multiple series starting/ending not at the same position could result in wrong zoom on `ValueAxis`.
- `DateAxis` with millisecond `baseInterval` sometimes could show one millisecond less than actual data point in its tooltip.

[Full change log](https://github.com/amcharts/amcharts4/blob/master/dist/es2015/CHANGELOG.md).

Make sure you clear your browser cache after upgrading. And feel free to contact us again if you are still experiencing this issue.
